linq4j/src/main/java/org/apache/calcite/linq4j/EnumerableDefaults.java

```diff
@@ -2636,6 +2636,56 @@ public static boolean isMergeJoinSupported(JoinType joinType) {
     return orderBy(source, keySelector, Collections.reverseOrder(comparator));
   }
 
+  /**
+   * A sort implementation based on a heap (or priority queue).
```

Comment:
I think this javadoc should mention that this sort implementation handles also fetch & offset, and what they mean.